test/hotspot/jtreg/serviceability/jvmti/vthread/ToggleNotifyJvmtiTest/ToggleNotifyJvmtiTest.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @summary Verifies JVMTI works for agents loaded into running VM
+ * @requires vm.jvmti
+ * @requires vm.continuations
+ * @enablePreview
+ * @library /test/lib /test/hotspot/jtreg
+ * @build jdk.test.whitebox.WhiteBox
+ *
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller jdk.test.whitebox.WhiteBox
+ * @run main/othervm/native -XX:+WhiteBoxAPI -Xbootclasspath/a:. -agentlib:ToggleNotifyJvmtiTest ToggleNotifyJvmtiTest
+ * @run main/othervm/native -XX:+WhiteBoxAPI -Xbootclasspath/a:. -Djdk.attach.allowAttachSelf=true ToggleNotifyJvmtiTest attach
+ */
+
+//import compiler.whitebox.CompilerWhiteBoxTest;
+import com.sun.tools.attach.VirtualMachine;
+import java.util.concurrent.ThreadFactory;
+import jdk.test.whitebox.WhiteBox;
+
+class TestedThread extends Thread {
+    private volatile boolean threadReady = false;
+    private volatile boolean shouldFinish = false;
+
+    // make thread with specific name
+    public TestedThread(String name) {
+        super(name);
+    }
+
+    // We will temporarily set a breakpoint on this method when the thread should be suspended.
+    // If we hit the breakpoint, then something is wrong.
+    public void breakpointCheck() {
+        return;
+    }
+
+    // run thread continuously
+    public void run() {
+        // run in a loop
+        threadReady = true;
+        int i = 0;
+        int n = 1000;
+        while (!shouldFinish) {
+            breakpointCheck();
+            if (n <= 0) {
+                n = 1000;
+                ToggleNotifyJvmtiTest.sleep(1);
+            }
+            if (i > n) {
+                i = 0;
+                n = n - 1;
```

Comment:
This code was copied originally from the vmTestbase to SuspendThread* tests and some other tests.
I can do all suggested simplifications but not sure if it is really necessary.
It does not matter what exactly the method does because it just simulates some thread activity.
Would it better to keep copy/pasted methods the same?

Reply:
ok